src/components/dataItem/StyleByDataItem.js

```diff
@@ -38,31 +38,39 @@ const StyleByDataItem = ({ program, programStage, error }) => {
 
     const dataItems = [ITEM_NONE, ...eventDataItems]
 
-    if (
-        styleDataItem &&
-        !dataItems.find((item) => item.id === styleDataItem.id)
-    ) {
-        return (
-            <Help error>
-                {i18n.t('Selected value not available in in list: {{name}}', {
-                    name: styleDataItem.name,
-                    nsSeparator: '^^',
-                })}
-            </Help>
+    const internalError =
+        styleDataItem && !dataItems.find((item) => item.id === styleDataItem.id)
+    let internalErrorText
+    if (internalError) {
+        internalErrorText = i18n.t(
+            'Previously selected value not available in list: {{id}}',
+            {
+                id: styleDataItem.id,
+                nsSeparator: '^^',
+            }
         )
     }
 
     return (
         <div>
             <SelectField
                 label={i18n.t('Style by data item')}
-                value={styleDataItem ? styleDataItem.id : null}
+                value={
+                    !(error || internalError)
+                        ? styleDataItem
```

Comment:
Could you extract the nested tertiary? I don't think it would pass a Code Climate check if it were enabled on this repo.